guava/src/main/scala/magnolify/guava/semiauto/FunnelDerivation.scala

```diff
@@ -62,7 +62,7 @@ trait FunnelImplicits {
   implicit val charSequenceFunnel: Funnel[CharSequence] = Funnels.unencodedCharsFunnel()
 
   implicit val booleanFunnel: Funnel[Boolean] = funnel[Boolean](_.putBoolean(_))
-  implicit val stringFunnel: Funnel[String] = funnel[String](_.putString(_, Charsets.UTF_8))
+  implicit val stringFunnel: Funnel[String] = Funnels.unencodedCharsFunnel.asInstanceOf[Funnel[String]]
```

Comment:
Guess instead of using the cast, this could also be `funnel[String](_. putUnencodedChars(_))`

Reply:
If we wrote our own typeclass for this (that was equivalent to `Funnel` but defined in Scala), we could use Scala's SAM support to define instances, like `implicit val stringFunnel: OurFunnel[String] = _ putUnencodedChars _`, add methods to it like `contramap` (which is what the `by` method in here seems to be?), `widen`, add a variance annotation to it, etc... which could then be converted to a Guava Funnel at use-site?

Reply:
Since `Funnel` is invariant in the type param, maybe we could define helpers like other libraries have, e.g.
```scala
def narrowFunnel[AA, A <: AA](funnel: Funnel[AA]): Funnel[A] = funnel.asInstanceOf[Funnel[A]]
```
so then `stringFunnel` could be defined as:
```scala
implicit val stringFunnel: Funnel[String] = narrowFunnel(Funnels.unencodedCharsFunnel)
```
?
And maybe even something to not need the cast on the "unboxed" funnels like Int, Long, etc...

Reply:
I think a simple solution would be to define a single implicit def for all `CharSequence` instances 
```scala
  implicit def charSequenceFunnel[T <: CharSequence]: Funnel[T] =
    Funnels.unencodedCharsFunnel().asInstanceOf[Funnel[T]]
 ```